checker/jdk/index/src/java/lang/Appendable.java

```diff
@@ -105,7 +105,7 @@
      * @throws  IOException
      *          If an I/O error occurs
      */
-    Appendable append(CharSequence csq, @IndexOrHigh("#1") int start, @IndexOrHigh("#1") int end) throws IOException;
+    Appendable append(CharSequence csq, @IndexFor("#1") int start, @IndexOrHigh("#1") int end) throws IOException;
```

Comment:
This was not a mistake. It is `@IndexOrHigh`, because `csq.length()` is a valid value for `start` if `end` is also `csq.length()`.  The case `start=csq.length()` is only invalid if `start>end`, which can happen for other values of `start` as well and it is a property which is not checked by the index checker.
On the other hand, `@IndexFor` would report false positives in code such as:
```java
void m(Appendable a, @MinLen(10) CharSequence csq){
  a.append(csq, 10, csq.length());
}
```

Reply:
You're right. I misunderstood the documentation, and I'll revert this change. Thanks for pointing this out.